src/python/pants/backend/python/dependency_inference/scripts/_pants_dep_parser/main.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+# -*- coding: utf-8 -*-
+
+# NB: This must be compatible with Python 2.7 and 3.5+.
+# NB: If you're needing to debug this, an easy way is to just invoke it on a file.
+#   E.g.
+#   $ export ROOT=src/python/pants/backend/python/dependency_inference/scripts/
+#   $ PYTHONPATH=$ROOT STRING_IMPORTS=y python $ROOT/_pants_dep_parser/main.py FILE
+#   Or
+#   $ ./pants --no-python-infer-imports run \
+#     src/python/pants/backend/python/dependency_inference/scripts/_pants_dep_parser/main.py \
+#     -- src/python/pants/base/specs.py
+
+from __future__ import print_function, unicode_literals
+
+import ast
+import importlib
+import json
+import os
+import sys
+from io import open
+
+from _pants_dep_parser.dependency_visitor_base import FoundDependencies
+
+
+def main(filename):
+    with open(filename, "rb") as f:
+        content = f.read()
+    try:
+        tree = ast.parse(content, filename=filename)
+    except SyntaxError:
+        return
+
+    package_parts = os.path.dirname(filename).split(os.path.sep)
+    visitor_classnames = os.environ.get(
+        "VISITOR_CLASSNAMES",
+        "_pants_dep_parser.general_dependency_visitor.GeneralDependencyVisitor",
```

Comment:
Just to check, this is here so we can run the script? Bummer we duplicate this info...

Reply:
The env var is so we can load the visitor classes at runtime.  It's not duplicate information - this is how the information is passed from the Pants code to the script.

The default is "duplicated" in a sense, for convenience when debugging.

Reply:
Yeah thats what I meant. I wonder if we could provide an opt-in goal/command that makes the debugging easy to do for people (even in their own repo).

Anywho...